File: Examples/Fibonacci/Fibonacci.java
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.0

import java.util.ArrayList;
import java.util.List;

public final class Fibonacci {
    private Fibonacci() {}

    private static List<Long> generate(int count) {
        var values = new ArrayList<Long>(count);
        long previous = 0;
        long current = 1;
        for (var index = 0; index < count; ++index) {
            values.add(previous);
            var next = previous + current;
            previous = current;
            current = next;
        }
        return List.copyOf(values);
    }

    public static void main(String[] args) {
        generate(12).forEach(System.out::println);
    }
}

File: Examples/Fibonacci/Fibonacci.cs
// SPDX-FileCopyrightText: 2026 Progmasoft <support@progmasoft.com>
// SPDX-License-Identifier: MPL-2.0 WITH AdditionRef-Progmasoft-Exception-1.1

using System;
using System.Collections.Generic;

namespace Examples.Fibonacci;

public static class Fibonacci
{
    private static IEnumerable<long> Generate(int count)
    {
        long previous = 0;
        long current = 1;

        for (var index = 0; index < count; ++index)
        {
            yield return previous;
            (previous, current) = (current, previous + current);
        }
    }

    public static void Main()
    {
        foreach (var value in Generate(12))
        {
            Console.WriteLine(value);
        }
    }
}
